Sources/OktaIdx/Internal/Implementations/Version1/Responses/Responses.swift

```diff
@@ -263,4 +263,62 @@ extension IDXClient.APIVersion1 {
         let refreshToken: String?
         let idToken: String?
     }
+
+    struct Redirect {
+        enum Query {
+            enum Key {
+                static let interactionCode = "interaction_code"
+                static let state = "state"
+                static let error = "error"
+                static let errorDescription = "error_description"
+            }
+            
+            enum Value {
+                static let interactionRequired = "interaction_required"
+            }
+        }
+        
+        let url: URL
+        let scheme: String
+        let path: String
+        
+        let interactionCode: String?
+        let state: String?
+        let error: String?
+        let errorDescription: String?
+        
+        let interactionRequired: Bool
+        
+        init?(url: String) {
+            guard let url = URL(string: url) else {
+                return nil
+            }
+            
+            self.init(url: url)
+        }
+
+        init?(url: URL) {
+            self.url = url
+            
+            guard let urlComponents = URLComponents(string: url.absoluteString), let scheme = urlComponents.scheme else {
+                return nil
+            }
+
+            self.scheme = scheme
+            self.path = urlComponents.path
+            
+            let queryItems = urlComponents.queryItems
+            
+            let queryValue: (String) -> String? = { name in
+                queryItems?.first { $0.name == name }?.value?.removingPercentEncoding
+            }
+
+            self.interactionCode = queryValue(Query.Key.interactionCode)
+            self.state = queryValue(Query.Key.state)
+            
+            self.error = queryValue(Query.Key.error)
+            self.errorDescription = queryItems?.first { $0.name == Query.Key.errorDescription }?.value?.removingPercentEncoding
+            self.interactionRequired = self.error == "interaction_required"
```

Comment:
Can you enclose the RHS of this expression in parenthesis, to make it clear that you're expecting a boolean result from this?

```suggestion
            self.interactionRequired = (self.error == "interaction_required")
```

Reply:
Done.